remix/app/routes/_app.newsletter.update.$id.test.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * @jest-environment node
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { Newsletter } from '@prisma/client';
+
+import { action } from './_app.newsletter.update.$id';
+import appHandler from '../lib/handler/app.handler';
+import { prismaMock } from '../tests/database';
+import { newsletterFactory } from '../tests/factories/newsletter.factory';
+import { userFactory } from '../tests/factories/user.factory';
+import { makeRequest } from '../tests/helpers/request';
+
+jest.mock('../config/auth.server', () => ({
+  authenticator: {
+    isAuthenticated: jest.fn(),
+  },
+}));
+
+jest.mock('../lib/handler/app.handler');
+
+describe('PUT /newsletter/update/:id', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('givens valid newsletter params with a creator user', () => {
```

Comment:
You still make the same typo 🙈 

```suggestion
  describe('given valid newsletter params', () => {
```

We should also wrap the block into a parent block for authenticated users since `callback(user)` is implemented for the  `appHandler`:

```javascript
describe('given the user is authnticated () => {
  describe('given valid newsletter params () => {
  })
})

describe('given the user is NOT authnticated () => {

})
```

Reply:
Umm, Sorry if I keep making the typo on testing, I know that we should write one test to expect one behavior only. In this case, I just want to provide more context on the description. 

But I know my test description still needs to improve, do you have any good source that shows how to write the test description?

Improved here: [d426473](https://github.com/nimblehq/js-framework-benchmark/pull/126/commits/d426473d63ea5e96d4c67a790b23169b39f78847)

Reply:
We don't have specific content on Compass for it yet (we should). So the cloest I can recommend is https://www.betterspecs.org/

Reply:
Thanks for the recommendation @olivierobert, will try to follow the practices from that website first.